---
title: "Phyloseq Plot General Commands"
author: "Erin Dahl"
date: "2/4/2020"
output: 
  html_document:
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: false
---

# R Markdown Document of Helpful Phyloseq Commands

## Function plot_bar()

The command __`plot_bar(phyloseqObject)`__ is used to plot summary graphics of differences in taxa abundance between samples using a phyloseq object.

The only required argument in this function is a physeq object which generally contains an OTU table, sample data (metadata), and a taxonomy table. It can also contain a phylo-class which is used for constructing a phylogeny tree.

An example of this function:

`plot_bar(phyloseqObject)`

`phyloseqObject` is the name of the phyloseq object that will be plotted.

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsBasic.png)


## Title

To add a title to the graph, use the argument `title = ` to indicates the title of the plot.

`plot_bar(phyloseqObject, title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample')`

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsTitle.png)

`title`  must be specified with and "=" when used as a parameters becuase it is not a required argument.

### Center Title

The default setting for plots is a left aligned title. In the example above the title is fairly long, and looks fine as is. 

If the title were much shorter, it may be helpful to center it using `plot_bar(phyloseqObject, title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample') + theme(plot.title = element_text(hjust = 0.5)`

## Filled Bars

To have filled bars on the plot, specify the fill argument. This sample variable is generally from the taxonomy table within the phyloseq object.
In the example, the fill selected was _Phylum_, however this could be changed to a different taxonomic rank such as _Kingdom_, _Genus_, etc. 

`fill` indicates which sample variable to be used to fill coloring of bars  

`plot_bar(phyloseqObject,fill='Phylum', title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample')`

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsFill.png)

## Y Limit

The `ylim()` function can be added to the function to set a y-axis limit. An x-axis `xlim()` is also avalible. In this example the y-axis is set a y-minimum of 0 and a y-maximum of 100.

`plot_bar(phyloseqObject,fill='Phylum', title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample') + ylim(c(0,100))`

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsYlim.png)

## Remove Black Lines in Fill

To remove the black lines inside the filled plot use `+ geom_bar(stat="identity")`.

`plot_bar(phyloseqObject,fill='Phylum', title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample') + ylim(c(0,100))+ geom_bar(stat="identity")`

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsNoFillLines.png)

## Remove Legend

To remove the legend use `+ theme(legend.position='none')` following the `plot_bar()` function.

`plot_bar(phyloseqObject,fill='Phylum', title = ' Proportion of Contaminant ASVs Not in the Blank Control Sample') + ylim(c(0,100))+ geom_bar(stat="identity") + theme(legend.position='none')`

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsNoLegend.png)

## Re-size Legend

 In this example, it makes the most sense to keep the legend, but to make the visual better, change the legend size. If the legend is disproportionate to the graph, following the `bar_plot()` function add `+ theme(legend.key.size = unit(leg_size, "cm"))` to scale the legend. Set the `leg_size` variable equal to 0.5 and adjust as needed.

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsResizeLegend.png)


## Color Palette

To change the color palette of the graph, include the RColorBrewer library. Find a [color palette](https://rdrr.io/cran/RColorBrewer/man/ColorBrewer.html) in this library to use. To change the color palette to blue for example, use `+ scale_fill_brewer(palette = "Blues")` after the `plot_bar()` function.

![](C:/Users/dahleri/Documents/RMarkdownTraining/PhyloseqPlotGeneralCommands/GeneralCommandsColor.png)

When changing the color palette, be aware that you may have more variables in the legend than in the amount of colors in the color palatte, In the case above, there were not enough unique colors in the palette to span each one of the Phylum in the legend.





